lib/components/element.rb

```diff
@@ -45,16 +47,25 @@ def self.element(name, multiple: false, &config)
         get_instance_variable(plural_name)
       end
     end
+
     # rubocop:enable Metrics/AbcSize
     # rubocop:enable Metrics/CyclomaticComplexity
     # rubocop:enable Metrics/MethodLength
     # rubocop:enable Metrics/PerceivedComplexity
 
+    attr_reader :nested_block
```

Comment:
What is this for?

Reply:
I introduced the nested_block attribute to be able to access the optional block during validation. This way validations can be conditional whether a block was given or not.

Reply:
Any further thoughts?

Reply:
Oh I see, that makes sense I think. And you can also validate the presence of a block? Let me think about the naming of the variable though, I struggled with that one myself when I first wrote this :)

Reply:
Currently there are two ways to interact with the given block:

1) You can access the block directly through the attribute reader "#nested_block"
2) You can check for its presence with the "#nested_block?" method

I was struggling with its naming too - any suggestions are very welcome ;)

Reply:
Aha, I basically thought you just renamed the `@yield` instance variable, but now I see that's not the case.

When do you need to access the block and not the captured content of the block? I.e. when do you need `nested_block` instead of what's in `@yield`?

Reply:
Let me rephrase.. Do we need the additional instance variable? I.e. both `@yield` and `@nested_block`, or would one suffice?

Reply:
I guess we can eliminate one instance variable. Until now it was only required to validate for the presence of the block via the `nested_block?` (or how it will be named) method. But lets work out the naming of the instance variable and the xxx? method first. Any naming suggestions/preferences?

Reply:
I think it depends on what we want to be able to do. If we only want to support conditional validations depending on if a block was given or not, we could just add:

```ruby
class Element
  def block_given? # or something
    @yield.present?
  end
end

class SomeComponent
  validates :something, if: :block_given?
end
```

And skip the reader. However, do we also want to support presence of the block? 

```ruby
class SomeComponent
  validates :block, presence: true
end
```

If so, we do need a reader, and a name for it. I don't think `yield` is very good since it's a reserved word in Ruby. One idea had before was `content` or some variation thereof, but I would prefer to use something that you wouldn't  also want to use as an attribute name. So maybe some variation on `block` is a good idea. Off the top of my head: 

- `content_block`
- `yielded_block`
- `captured_block`
- `block`

Reply:
I just reviewed our use cases and they seem to match with your ideas quite nicely 👍 

We often used the basic presence validation on the `attr_reader`

```ruby
class SectionComponent < ApplicationComponent
  validates :content_block, presence: true
end
```
and sometimes had more "complex" validations using the `content_block?` method

```ruby
class OtherComponent < ApplicationComponent
  attribute :text

  validate :validate_content_is_present

  def validate_content_is_present
    return if content_block? || text.present?

    errors.add(:base, 'no block or text attribute is given!')
  end
end
```
Also the `content_block?`method was used in several component view templates to modify their rendering.

Guess we implement both then?

BTW I think naming it `content_block` really is an improvement. Nice one!